src/libs/Navigation/AppNavigator/createCustomStackNavigator/index.tsx

```diff
@@ -61,20 +61,23 @@ function ResponsiveStackNavigator(props: ResponsiveStackNavigatorProps) {
     const {stateToRender, searchRoute} = useMemo(() => {
         const routes = reduceCentralPaneRoutes(state.routes);
 
-        const lastRoute = routes[routes.length - 1];
-        const isLastRouteSearchRoute = getTopmostCentralPaneRoute({routes: [lastRoute]} as State<RootStackParamList>)?.name === SCREENS.SEARCH.CENTRAL_PANE;
+        // On narrow layout, if we are on /search route we want to hide the search central pane route.
+        if (isSmallScreenWidth) {
+            const searchCentralPaneIndex = routes.findIndex((route) => {
+                if (route.name !== NAVIGATORS.CENTRAL_PANE_NAVIGATOR) {
+                    return false;
+                }
 
-        const firstRoute = routes[0];
-
-        // On narrow layout, if we are on /search route we want to hide all central pane routes and show only the bottom tab navigator.
-        if (isSmallScreenWidth && isLastRouteSearchRoute) {
+                return (route.params && 'screen' in route.params && route.params.screen === SCREENS.SEARCH.CENTRAL_PANE) || route.state?.routes.at(-1)?.name === SCREENS.SEARCH.CENTRAL_PANE;
+            });
+            const filteredRoutes = searchCentralPaneIndex !== -1 ? [...routes.slice(0, searchCentralPaneIndex), ...routes.slice(searchCentralPaneIndex + 1)] : [...routes];
             return {
                 stateToRender: {
                     ...state,
-                    index: 0,
-                    routes: [firstRoute],
+                    index: filteredRoutes.length - 1,
+                    routes: filteredRoutes,
                 },
-                searchRoute: lastRoute,
+                searchRoute: routes[searchCentralPaneIndex],
```

Comment:
@luacmartins The reason I don't use `filter` is the `searchRoute` here expects that we pass the found search central pane route, so I need the found index of the route. The `searchRoute` is being used here
https://github.com/Expensify/App/blob/a451de460c45611f64bcd06961afb78b4e2f3c9b/src/libs/Navigation/AppNavigator/createCustomStackNavigator/index.tsx#L100

but I can't see any difference though with or without `searchRoute`. Maybe we can ask someone from SWM to check this too. If we can safely remove it, then we can use `filter`.